src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinResult = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
+        joinResult += separator;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
+        continue;
+      }
+
+      joinResult += this[i];
```

Comment:
You're calling `this[i]` three times, so it would be cool to create a variable for that and reuse it.